### PR TITLE
Add iphone se support

### PR DESCRIPTION
**Pull request by i2amsam**
This is an alternate implementation to https://github.com/fastlane/fastlane/pull/3980 which adds some more testing and pulls apart the filename grabbing so it's a bit more manageable for the different variations. 

image

### CODE REVIEW

File: frameit/lib/frameit/offsets.rb [hunk 1]
```diff
@@ -52,10 +59,17 @@ def self.image_offset(screenshot)
...
-            'offset' => "+201+48",
-            'width' => 970
-          }
+          if Frameit.config[:use_legacy_iphone5s]
+            return {
+              'offset' => "+201+48",
+              'width' => 970
+            }
+          else
+            {
```
Comment (mfurtak) [inline frameit/lib/frameit/offsets.rb]: missing a `return` compared to other places?

File: frameit/lib/frameit/options.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module Frameit
+  class Options
+    def self.available_options
+      @options ||= [
+        FastlaneCore::ConfigItem.new(key: :use_legacy_iphone5s,
+                                     env_name: "FRAMEIT_USE_LEGACY_IPHONE_5_S",
```
Comment (mfurtak) [inline frameit/lib/frameit/options.rb]: Nit: I'd expect `FRAMEIT_USE_LEGACY_IPHONE5S` given the name of the key

File: frameit/lib/frameit/template_finder.rb [hunk 3]
```diff
@@ -1,38 +1,82 @@
...
-          screenshot.device_name,
-          (screenshot.color == 'SpaceGray' ? "Space-Gray" : "Silver"),
-          (screenshot.orientation_name == "Horz" ? "horizontal" : "vertical")
-        ]
-        joiner = "-"
-      end
+
+      transformer = DEVICE_TO_TRANSFORM_TYPE[screenshot.device_name]
+
+      transformer ||= Type1Transform.new(screenshot.device_name)
```
Comment (mfurtak) [inline frameit/lib/frameit/template_finder.rb]: In what cases would we fail to find a proper transform in our lookup table? Why is falling back to a Type1 correct, rather than raising an error? Can that rationale be documented?

File: frameit/lib/frameit/template_finder.rb [hunk 4]
```diff
@@ -1,38 +1,82 @@
...
+      filename = transformer.transform(screenshot.color, screenshot.orientation_name)
 
       templates_path = [ENV['HOME'], FrameConverter::FRAME_PATH].join('/')
-      templates = Dir["../**/#{parts.join(joiner)}*.{png,jpg}"] # local directory
-      templates += Dir["#{templates_path}/**/#{parts.join(joiner)}*.{png,jpg}"] # ~/.frameit folder
+      templates = Dir["../**/#{filename}.{png,jpg}"] # local directory
+      templates += Dir["#{templates_path}/**/#{filename}.{png,jpg}"] # ~/.frameit folder
```
Comment (mfurtak) [inline frameit/lib/frameit/template_finder.rb]: There have been other places in the codebase where we've been bitten by varying capitalization on file extensions (.PNG vs. .png, etc.), and so have introduced `File::FNM_CASEFOLD` to calls to `Dir.glob()`. Consider whether that would be appropriate here.

File: frameit/spec/template_finder_spec.rb [hunk 5]
```diff
@@ -0,0 +1,110 @@
+require 'ostruct'
+describe Frameit do
+  describe Frameit::TemplateFinder do
+    describe 'it can find some screenshots' do
+      before do
+        ENV['HOME'] = '/thisdirectorydoesnotexist'
+      end
```
Comment (mfurtak) [inline frameit/spec/template_finder_spec.rb]: Does there need to be an equivalent after to restore this to a correct value?

**Comment by mfurtak** [comment]
I like what's going on with the the transform strategy extraction 👍 

If it's not too much work, I think it would be great for testability to allow the `TemplateFinder` to have it's base directory be provided at construction time (perhaps defaulting to  `ENV['HOME']`), so that the `ENV['HOME']` doesn't need to be globally adjusted for the tests.

File: frameit/spec/template_finder_spec.rb [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        })
+        expected_result = '../frameit/spec/fixtures/mock_frames/iPad Pro/iPad-Pro-Space-Gray-vertical.png'
+
+        expect(Frameit::TemplateFinder.get_template(screenshot)).to eq(expected_result)
+      end
+
+      it 'warns for an iPhone-SE if there is no template' do
+        screenshot = make_screenshot({
+          device_name: 'iPhone-SE',
+          color:  'SpaceGray',
```
Comment (mfurtak) [inline frameit/spec/template_finder_spec.rb]: nit: extra spacing before the value of `color:` here and a few other places

File: frameit/lib/frameit/options.rb (hunk 2 above)
Comment (i2amsam) [inline frameit/lib/frameit/options.rb]: I thought the `5S` was too close to `55`

File: frameit/lib/frameit/template_finder.rb (hunk 4 above)
Comment (i2amsam) [inline frameit/lib/frameit/template_finder.rb]: We control creating these files so we should be good.

File: frameit/lib/frameit/frame_converter.rb [hunk 7]
```diff
@@ -64,6 +64,23 @@ def convert_frames
 
         UI.important "Converting PSD file '#{psd}'"
         image = MiniMagick::Image.open(psd)
+
+        if psd =~ /iPhone-SE/
+          UI.success "Removing white background 🚫 ⬜️"
+
+          good_layers = image.layers.reject do |layer|
+            label = layer.details['Properties']['label']
+            label =~ /White B/i || label.to_s.length == 0
```
Comment (asfalcone) [inline frameit/lib/frameit/frame_converter.rb]: Is this rejecting layers with no labels? 

Maybe a comment in the code on how this works would be helpful

**Comment by asfalcone** [comment]
LGTM. Squash and merge! 🎃

File: frameit/lib/frameit/frame_converter.rb [hunk 8]
```diff
@@ -64,6 +64,27 @@ def convert_frames
...
+        if psd =~ /iPhone-SE/
+          UI.success "Removing white background 🚫 ⬜️"
+
+          # The iPhone-SE screenshots from April 2016 have
+          # 3 layers, a background, a product, and the 'put your image here' layer
+          # imagemagick seems to add an additional layer with no label which this the
+          # composite of all three.  We want to remove the background and composite layer
+          good_layers = image.layers.reject do |layer|
+            label = layer.details['Properties']['label']
+            label =~ /White B/i || label.to_s.length == 0
```
Comment (mfurtak) [inline frameit/lib/frameit/frame_converter.rb]: might want to reverse the order of these comparisons in case the label is ever `nil`

File: frameit/lib/frameit/frame_converter.rb [hunk 9]
```diff
@@ -64,6 +64,27 @@ def convert_frames
...
+          end
+          product_layer = good_layers.shift
+
+          good_layers.each do |layer|
+            product_layer.layers << layer
+          end
```
Comment (mfurtak) [inline frameit/lib/frameit/frame_converter.rb]: can you `+=` the rest of the `good_layers` onto `layers` in one go?

Comment (mfurtak) [inline frameit/lib/frameit/frame_converter.rb]: The answer, I was told, is no 😞

**Comment by mfurtak** [comment]
:ship: it!

File: frameit/bin/frameit [hunk 10]
```diff
@@ -19,6 +19,7 @@ class FrameItApplication
     program :help_formatter, :compact
 
     global_option('--verbose') { $verbose = true }
+    FastlaneCore::CommanderGenerator.new.generate(Frameit::Options.available_options)
```
Comment (KrauseFx) [inline frameit/bin/frameit]: Why is this necessary?

**Comment by KrauseFx** [comment]
Great tests, great PR!! 👍

File: frameit/bin/frameit (hunk 10 above)
Comment (mfurtak) [inline frameit/bin/frameit]: frameit previously had no `ConfigItem` options, but now it has one. This line is required to turn the `ConfigItem`s into `global_option`s in the `Commander` context.

**Comment by felix-dumit** [comment]
If I run frameit directly it is working fine, however if I run frameit through fastlane I get an error when trying to add the iPhone SE frames.

Here is a portion of the log when running `fastlane run frameit --verbose`:

```
INFO [2016-05-05 13:23:31.72]: Framing screenshots at path ./fastlane/
ERROR [2016-05-05 13:23:31.75]: undefined method `[]' for nil:NilClass
ERROR [2016-05-05 13:23:31.75]: Backtrace:
	/Users/felixdumit/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/frameit-2.6.1/lib/frameit/screenshot.rb:29:in `device_name'
	/Users/felixdumit/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/frameit-2.6.1/lib/frameit/screenshot.rb:54:in `mac?'
	/Users/felixdumit/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/frameit-2.6.1/lib/frameit/screenshot.rb:73:in `frame!'
	/Users/felixdumit/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/frameit-2.6.1/lib/frameit/runner.rb:29:in `block in run'
```

**Comment by adamhrz** [comment]
@felix-dumit I ran into this and needed to update fast lane to 1.87.1 which includes better bathing support for actions
